lib/firehose/server/consumer.rb

```diff
@@ -0,0 +1,18 @@
+module Firehose
+  module Server
+    # Represents the current state of a consumer to the Firehose
+    # server and middleware.
+    class Consumer
```

Comment:
I think the state managed by this class should probably just be moved into the `Channel` class. The two are so tightly intertwined at this point that there doesn't seem to be a benefit to keeping this separate. Even in the multiplexing code you are creating a new `Channel` and `Consumer` object for each subscription.